packages/lab/src/list-next/ListNext.tsx

```diff
@@ -16,14 +18,19 @@ import { ListNextContext } from "./ListNextContext";
 
 const withBaseName = makePrefixer("saltListNext");
 
-export interface ListNextProps extends HTMLAttributes<HTMLUListElement> {
+export interface ListNextProps
+  extends Omit<ComponentPropsWithoutRef<"ul">, "onChange"> {
   /**
    * If true, all items in list will be disabled.
    */
   disabled?: boolean;
-  /* Value for the uncontrolled version. */
+  /* Value for the controlled version. */
+  highlightedIndex?: number;
```

Comment:
Is there a logic as to why highlighted is controlled by the index, but selected is controlled by the value of the relevant item? Maybe these should be aligned?

One potential issue is you might want to control a list using the `onChange` prop of `List` (in the current example you use the `onClick` of `ListItem`). But the `onChange` event of `List` doesn't pass the index of the clicked item as part of the callback data. If we do keep it as an index then at least we should also pass the clicked index in that callback to allow you to do something like:

```
const handleChange = (_, { value, index }) => {
    setSelectedItem(value);
    setHighlightedIndex(index);
  };
```
but I do think it would be nicer if you could simply do:
```
const handleChange = (_, { value }) => {
    setSelectedItem(value);
    setHighlightedItem(value);
  };
```

Reply:
Having looked a bit more it seems like it wouldn't be possible to use the onChange callback for this since it only triggers when a different item is selected.